File: guests/java/dev/kaya/milestone2kt/Progress.java
package dev.kaya.milestone2kt;

import dev.kaya.KayaApp;

/**
 * The progress conformance scene from the JVM. See
 * guests/rust/progress.rs and tools/scenes/progress.steps.
 */
final class Progress {
    static void app() {
        KayaApp app = new KayaApp();

        app.build(tx -> {
            tx.window(0).title("progress");
            tx.mount(tx.column(() -> {
                tx.progress(0.25); // progress#0
                tx.progressIndeterminate(); // progress#1
            }));
            return null;
        });

        app.dispatchLoop();
    }

    private Progress() {}
}

File: guests/go/progress/progress.go
// The progress conformance scene, Go port. See
// guests/rust/progress.rs and tools/scenes/progress.steps.
package progress

import (
	kaya "dev.kaya/bindings/go"
)

func App() *kaya.App {
	app := kaya.NewApp()

	app.Build(func(tx *kaya.Tx) {
		tx.Window(0).Title("progress")
		tx.Mount(tx.Column(func() {
			tx.Progress(0.25)                // progress#0
			tx.Progress(0).Indeterminate()   // progress#1
		}))
	})

	return app
}
